src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java

```diff
@@ -244,14 +238,119 @@ SecurityAnalysisReport runSync(SecurityAnalysisParameters securityAnalysisParame
+
+        if (lfParameters.getConnectedComponentMode().equals(LoadFlowParameters.ConnectedComponentMode.ALL)) {
+            List<LfNetwork> networkToSimulate = networks.getList().stream()
+                    .filter(n -> n.getNumCC() != ComponentConstants.MAIN_NUM && n.getValidity().equals(LfNetwork.Validity.VALID)).toList();
+            for (LfNetwork n : networkToSimulate) {
+                SecurityAnalysisResult resultOtherComponent = runSimulations(n, propagatedContingencies, parameters, securityAnalysisParameters, operatorStrategies, actions, limitReductions);
+
+                // Merge into first result
+                // PreContingency results first
+                preContingenctViolations.addAll(resultOtherComponent.getPreContingencyResult().getLimitViolationsResult().getLimitViolations());
+                mergedPreContingencyNetworkResult =
+                        mergeNetworkResult(resultOtherComponent.getPreContingencyResult().getNetworkResult(), mergedPreContingencyNetworkResult);
+
+                // PostContingency and OperatorStrategies results
+                mergeSecurityAnalysisResult(resultOtherComponent, postContingencyResults, operatorStrategyResults);
+            }
+        }
+        PreContingencyResult mergedPrecontingencyResult =
+                new PreContingencyResult(result.getPreContingencyResult().getStatus(),
+                        new LimitViolationsResult(preContingenctViolations),
+                        mergedPreContingencyNetworkResult);
+        return new SecurityAnalysisResult(mergedPrecontingencyResult, postContingencyResults, operatorStrategyResults);
+    }
+
+    void mergeSecurityAnalysisResult(SecurityAnalysisResult resultToMerge, List<PostContingencyResult> postContingencyResults,
+                                     List<OperatorStrategyResult> operatorStrategyResults) {
+        resultToMerge.getPostContingencyResults().forEach(postContingencyResult -> {
+            Optional<PostContingencyResult> optOriginalResult = postContingencyResults.stream()
+                    .filter(originalPostContingencyResult -> originalPostContingencyResult.getContingency().getId().equals(postContingencyResult.getContingency().getId()))
+                    .findAny();
+
+            if (optOriginalResult.isPresent()) {
+                PostContingencyResult originalResult = optOriginalResult.get();
+                NetworkResult mergedNetworkResult = mergeNetworkResult(postContingencyResult.getNetworkResult(), originalResult.getNetworkResult());
+                List<LimitViolation> violations = new ArrayList<>(postContingencyResult.getLimitViolationsResult().getLimitViolations());
+                violations.addAll(originalResult.getLimitViolationsResult().getLimitViolations());
+
+                PostContingencyResult mergedPostContingencyResult =
+                        new PostContingencyResult(originalResult.getContingency(), originalResult.getStatus(),
```

Comment:
if original status is different from the current contigency result (eg one converged the other failed), since you cannot report this with the current API, a warning should be logged (and maybe something reported to be able show this to an application user ?)
The best solution would however to have the right result API.

Reply:
Yes, LoadFlowResult can be an inspiration on that.